examples/demo/src/reviews/ReviewCreate.tsx

```diff
@@ -0,0 +1,69 @@
+import * as React from 'react';
+import {
+    SimpleForm,
+    Create,
+    ReferenceInput,
+    TextInput,
+    DateInput,
+    AutocompleteInput,
+    required,
+    useNotify,
+    useRedirect,
+    getRecordFromLocation,
+} from 'react-admin';
+
+import StarRatingInput from './StarRatingInput';
+import { useLocation } from 'react-router';
+
+const ReviewCreate = () => {
+    const notify = useNotify();
+    const redirect = useRedirect();
+    const location = useLocation();
+
+    const onSuccess = (_: any) => {
+        const record = getRecordFromLocation(location);
```

Comment:
Why do you get the record from the location and not the record from the onSuccess callback parameter?

Reply:
I used the location in order to keep the possibility to create straightly reviews. If I use the parameter of the callback I do not know from where I come from and the redirection works in all cases.

Reply:
In short: to handle cases where you don't come from a product edition view

Reply:
ah, I get it: the record isn't the created `review`, but a `product`. To clarify, you should perhaps name the variable `product`.